feat(Translation): Convert step title input to translatable

geoffreykwan opened the pull request:
## Changes
- Convert step title input to translatable input

## Test
- Make sure the step title is translatable

geoffreykwan committed:
feat(Translation): Convert step title input to translatable

hirokiterashima reviewed (changes requested):
1. Can we show the english text under the title like we do with other inputs?
<img width="515" alt="Screenshot 2024-07-02 at 10 35 41 AM" src="https://github.com/WISE-Community/WISE-Client/assets/119416/4634aaa7-4f36-44c7-b085-1be6150d2eb8">

2. The ```Step Title {{ nodePosition }}:``` and ``` Lesson Title {{ nodePosition }}:``` i18n strings appear to be removed from messages.xlf. Shouldn't they be translatable?

geoffreykwan committed:
Merge branch 'issue-1513-multiple-languages-per-unit' into convert-step-title-input-to-translatable

geoffreykwan committed:
feat(Translation): Fix step title authoring styling. Fix translations.

geoffreykwan committed:
fix(Translation): Fix width of translatable inputs in default language

geoffreykwan commented:
The English text now shows when authoring other languages. I moved the translations from the .html into the .ts file. I also fixed an issue with the width for all translatable inputs.

hirokiterashima commented on src/assets/wise5/authoringTool/node/edit-node-title/edit-node-title.component.ts:
```diff
@@ -4,21 +4,23 @@ import { TeacherProjectService } from '../../../services/teacherProjectService';
-    this.isGroupNode = this.node.isGroup();
     this.nodeJson = this.projectService.getNodeById(this.node.id);
-    this.nodePosition = this.projectService.getNodePositionById(this.node.id);
+    this.isGroupNode = this.node.isGroup();
+    this.label =
+      (this.isGroupNode ? $localize`Lesson Title` : $localize`Step Title`) +
+      ' ' +
+      this.projectService.getNodePositionById(this.node.id);
```
Remove isGroupNode class variable and use this.node.isGroup() directly in the ternary expression?
```suggestion
    this.label =
      (this.node.isGroup() ? $localize`Lesson Title` : $localize`Step Title`) +
      ' ' +
      this.projectService.getNodePositionById(this.node.id);
```

hirokiterashima reviewed (approved):
LGTM. I think you can remove the isGroupNode class variable.

geoffreykwan committed:
refactor(Translation): Remove unnecessary variable

geoffreykwan committed:
Merge branch 'issue-1513-multiple-languages-per-unit' into convert-step-title-input-to-translatable
